### ISSUE TITLE
Some floating images not being output to pdf since 0.42

### ISSUE THREAD

**Issue by jessejoe** [Kozea/WeasyPrint#576]
Some float images that display fine in HTML are not being included in rendered WeasyPrint PDFs using `0.42`+.

## Reproduce

A very simple self-contained html document which has a couple images. This is a simplified version using actual CSS from [Bootstrap](https://getbootstrap.com) (which is what my actual site is using):
```html
<html>
<body>
<img src="https://i.imgur.com/mK3ufI3.png">
<div style="float: right;"><img style="max-width: 100%;" src="https://i.imgur.com/mK3ufI3.png"></div>
</body>
</html>
```

The output looks like this (or live https://jsfiddle.net/kzLhtejh/):
![image](https://user-images.githubusercontent.com/1694586/36186957-46cf5a34-1110-11e8-8724-e2a298c112aa.png)

When generating a PDF from this HTML with WeasyPrint `0.41`, the render looks correct:
```
$ pip install WeasyPrint==0.41
Collecting WeasyPrint==0.41
  Using cached WeasyPrint-0.41-py3-none-any.whl
...
Installing collected packages: WeasyPrint
Successfully installed WeasyPrint-0.41

$ python -c "from weasyprint import HTML; HTML('pdf.html').write_pdf('test.pdf')"
$
```

PDF render:
![image](https://user-images.githubusercontent.com/1694586/36187003-9c7e07f0-1110-11e8-9c46-76b3f7a0cd0e.png)

However when simply upgrading to WeasyPrint `0.42`, the PDF render is missing the floated image:
```
$ pip install WeasyPrint==0.42
Collecting WeasyPrint==0.42
  Using cached WeasyPrint-0.42-py3-none-any.whl
...
Installing collected packages: WeasyPrint
  Found existing installation: WeasyPrint 0.41
    Uninstalling WeasyPrint-0.41:
      Successfully uninstalled WeasyPrint-0.41
Successfully installed WeasyPrint-0.42

$ python -c "from weasyprint import HTML; HTML('pdf.html').write_pdf('test.pdf')"
$
```

PDF render:
![image](https://user-images.githubusercontent.com/1694586/36187072-e3b7424e-1110-11e8-8b9b-aa14e5a863a3.png)

All subsequent versions of weasyprint have the same result. Here is a comparison of the two versions, there were quite a few changes to float logic - https://github.com/Kozea/WeasyPrint/compare/v0.41...v0.42

I did notice #561, however cairo versions do not have an affect, it's purely in WeasyPrint.

**Comment by jessejoe** [comment]
A little more research...

Commit 1076b106ea212c3ddc15730a6efda30f16c527e9 seems to have broken this.

If I reset the repo to this commit and install it:
```
$ git reset --hard 1076b106ea212c3ddc15730a6efda30f16c527e9
HEAD is now at 1076b106 Ignore intrinsic size for replaced boxes with percentage-based width

$ pip install -e .
Obtaining file:///Users/vagrant/jesse/WeasyPrint
...
Installing collected packages: WeasyPrint
  Running setup.py develop for WeasyPrint
Successfully installed WeasyPrint-0.41
```

The generated PDF is still broken. If I revert this commit:
```
$ git reset --hard HEAD~1
HEAD is now at a317b636 Fix typo
```

The generated PDF is correct.

Note that 1076b106ea212c3ddc15730a6efda30f16c527e9 says it was implemented to fix #227.

**Comment by liZe** [comment]
Thanks for the example, and sorry for the regression!
